File: computer-vision/src/api/routes/cameras.py
from fastapi import APIRouter, HTTPException, Depends, Query
from typing import List, Dict
from ..models.camera import Camera, CameraCreate, CameraUpdate
import json
from pathlib import Path

router = APIRouter()

def load_camera_configs():
    config_path = Path("/Users/khalidll/Desktop/project-eagle/config/camera-config.json")
    try:
        with open(config_path, 'r') as f:
            return json.load(f)
    except Exception as e:
        print(f"Error loading camera configs: {e}")
        return {"cameras": []}

@router.get("/cameras", response_model=List[Camera])
async def list_cameras(client_id: str = Query(..., description="Client ID to filter cameras")) -> List[Camera]:
    """
    List all available cameras for a specific client
    """
    configs = load_camera_configs()
    client_cameras = [
        Camera(
            id=cam['camera_id'],
            name=f"Camera {cam['camera_id']}",
            description=f"Camera in {cam['zone']} zone",
            rtsp_url=cam['rtsp_url'],
            kvs_stream_id=cam.get('kvs_stream_id'),
            status="active" if cam['enabled'] else "inactive"
        )
        for cam in configs['cameras']
        if cam['client_id'] == client_id
    ]
    return client_cameras

@router.get("/cameras/{camera_id}", response_model=Camera)
async def get_camera(camera_id: str) -> Camera:
    """
    Get specific camera details
    """
    # TODO: Implement camera retrieval logic
    raise HTTPException(status_code=404, detail="Camera not found")

File: frontend/src/app/api/cameras.ts
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8000/api/v1';

export interface Camera {
  id: string;
  name: string;
  description: string;
  rtsp_url: string;
  kvs_stream_id?: string;
  status: 'active' | 'inactive';
}

export const cameraApi = {
  /**
   * Fetch all cameras for a specific client
   * @param clientId - The ID of the client to fetch cameras for
   */
  getCameras: async (clientId: string): Promise<Camera[]> => {
    try {
      const response = await axios.get(`${API_BASE_URL}/cameras`, {
        params: { client_id: clientId }
      });
      return response.data;
    } catch (error) {
      console.error('Error fetching cameras:', error);
      throw error;
    }
  },

  /**
   * Fetch a specific camera by ID
   * @param cameraId - The ID of the camera to fetch
   */
  getCamera: async (cameraId: string): Promise<Camera> => {
    try {
      const response = await axios.get(`${API_BASE_URL}/cameras/${cameraId}`);
      return response.data;
    } catch (error) {
      console.error(`Error fetching camera ${cameraId}:`, error);
      throw error;
    }
  }
}; 